### PR TITLE
Ebtree fixes

### PR DESCRIPTION
**Pull request by davisp**
Fixes for ebtree

### CODE REVIEW

File: src/ebtree/src/ebtree.erl [hunk 1]
```diff
@@ -411,6 +419,9 @@ range(Db, #tree{} = Tree, StartKey, EndKey, AccFun, Acc0) ->
     end).
 
 
+range(_Tx, #tree{}, #node{level = 0, members = []}, _StartKey, _EndKey, _AccFun, Acc0) ->
```
Comment (rnewson) [inline src/ebtree/src/ebtree.erl:422]: when will a leaf node have zero members? That... should never happen.

File: src/ebtree/src/ebtree.erl [hunk 2]
```diff
@@ -282,6 +283,9 @@ reduce(Db, #tree{} = Tree, StartKey, EndKey, Options) ->
     do_reduce(Tree, MapValues, ReduceValues).
 
 
+do_reduce(#tree{} = Tree, [], []) ->
+    reduce_values(Tree, [], false);
```
Comment (rnewson) [inline src/ebtree/src/ebtree.erl:287]: shouldn't that be `true`? the lack of map values is the signal we're working with reductions.

File: src/ebtree/src/ebtree.erl (hunk 1 above)
Comment (davisp) [reply src/ebtree/src/ebtree.erl:422]: When the tree is empty like it says in the commit message.

File: src/ebtree/src/ebtree.erl (hunk 2 above)
Comment (davisp) [reply src/ebtree/src/ebtree.erl:287]: We're not working with reductions though. We're reducing an empty list of key/value pairs. Plainly the empty list from the second argument is the same empty list being passed to reduce_values.

Also, I think most folks writing reduce functions would do better to think of it as "reduce empty list of key/values" vs "rereduce empty list of reduce values that might need to be guarded against for the normal case when you actually have reduce values".

File: src/ebtree/src/ebtree.erl (hunk 1 above)
Comment (rnewson) [reply src/ebtree/src/ebtree.erl:422]: got it. can you add a test that shows the problem pls.

File: src/ebtree/src/ebtree.erl (hunk 2 above)
Comment (rnewson) [reply src/ebtree/src/ebtree.erl:287]: kk

File: src/ebtree/src/ebtree.erl (hunk 1 above)
Comment (rnewson) [reply src/ebtree/src/ebtree.erl:422]: also tighten this to match only the root node (instead of `level=0`)

**Pull request merged**